app/models/event.rb

```diff
@@ -41,6 +41,7 @@ class Event < ActiveRecord::Base
   scope :canceled, -> { where(state: 'canceled') }
   scope :withdrawn, -> { where(state: 'withdrawn') }
   scope :highlighted, -> { where(is_highlight: true) }
+  scope :current, -> { joins(:event_type).where('start_time <= ? AND state = ?', Time.current, 'confirmed').map { |e| e if e.end_time > Time.current }.compact }
```

Comment:
I am struggling to write this as a scope with the new associations. If you have any tips, I would appreciate. Otherwise, I'll write a class method for now.